pkg/manifests/external_dns.go

```diff
@@ -347,6 +428,21 @@ func newExternalDNSDeployment(conf *config.Config, externalDnsConfig *ExternalDn
 
 	serviceAccount := externalDnsConfig.serviceAccountName
 
+	deploymentArgs := []string{
+		"--provider=" + externalDnsConfig.provider.string(),
+		"--interval=" + conf.DnsSyncInterval.String(),
+		"--txt-owner-id=" + conf.ClusterUid,
+		"--txt-wildcard-replacement=" + txtWildcardReplacement,
+	}
+
+	for deploymentArg := range externalDnsConfig.resourceTypes {
+		if externalDnsConfig.resourceTypes[deploymentArg] {
```

Comment:
what does this do?

Reply:
ensures the value is both present and set to true, vs somehow being set to false. I see how the variable name deploymentArg is confusing though I can rename

Reply:
use map[type]struct{} instead of map[type]bool for sets

Reply:
done